compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1728,8 +1728,8 @@ rustc_queries! {
     query upvars_mentioned(def_id: DefId) -> Option<&'tcx FxIndexMap<hir::HirId, hir::Upvar>> {
         desc { |tcx| "collecting upvars mentioned in `{}`", tcx.def_path_str(def_id) }
     }
-    query maybe_unused_trait_import(def_id: LocalDefId) -> bool {
-        desc { |tcx| "maybe_unused_trait_import for `{}`", tcx.def_path_str(def_id.to_def_id()) }
+    query maybe_unused_trait_imports(_: ()) -> &'tcx rustc_data_structures::fx::FxIndexSet<LocalDefId> {
```

Comment:
You can put a `use` statement in `rustc_middle::ty::query`.
Returning a reference is the right thing to do here. Queries can only return `Copy` types anyway.